src/main/java/com/github/technus/tectech/thing/metaTileEntity/multi/godforge_modules/GT_MetaTileEntity_EM_PlasmaModule.java

```diff
@@ -0,0 +1,221 @@
+    private int fusionTier = 0;
+    private int inputMaxParallel = 0;
+
+    public GT_MetaTileEntity_EM_PlasmaModule(int aID, String aName, String aNameRegional) {
+        super(aID, aName, aNameRegional);
+    }
+
+    public GT_MetaTileEntity_EM_PlasmaModule(String aName) {
+        super(aName);
+    }
+
+    @Override
+    public IMetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+        return new GT_MetaTileEntity_EM_PlasmaModule(mName);
+    }
+
+    long wirelessEUt = 0;
+
+    @Override
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic() {
+
+            @NotNull
+            @Override
+            protected CheckRecipeResult validateRecipe(@Nonnull GT_Recipe recipe) {
+                wirelessEUt = (long) recipe.mEUt * getMaxParallel();
+                if (getUserEU(userUUID).compareTo(BigInteger.valueOf(wirelessEUt * recipe.mDuration)) < 0) {
+                    return CheckRecipeResultRegistry.insufficientPower(wirelessEUt * recipe.mDuration);
+                }
+                if (recipe.mSpecialValue > fusionTier
+                        || Objects.equals(recipe.mSpecialItems.toString(), "true") && !multiStep) {
+                    return SimpleCheckRecipeResult.ofFailure("missing_upgrades");
+                }
+                return CheckRecipeResultRegistry.SUCCESSFUL;
+            }
+
+            @NotNull
+            @Override
+            protected CheckRecipeResult onRecipeStart(@Nonnull GT_Recipe recipe) {
+                wirelessEUt = (long) recipe.mEUt * maxParallel;
+                if (!addEUToGlobalEnergyMap(userUUID, -calculatedEut * duration)) {
+                    return CheckRecipeResultRegistry.insufficientPower(wirelessEUt * recipe.mDuration);
+                }
+                currentParallel = calculatedParallels;
+                EUt = calculatedEut;
+                setCalculatedEut(0);
+                return CheckRecipeResultRegistry.SUCCESSFUL;
+            }
+
+            @Nonnull
+            @Override
+            protected GT_OverclockCalculator createOverclockCalculator(@Nonnull GT_Recipe recipe) {
+                return super.createOverclockCalculator(recipe).setEUt(TierEU.MAX);
+            }
+        };
+    }
+
+    @Override
+    protected void setProcessingLogicPower(ProcessingLogic logic) {
+        logic.setAvailableVoltage(Long.MAX_VALUE);
+        logic.setAvailableAmperage(Integer.MAX_VALUE);
+        logic.setAmperageOC(false);
+        logic.setMaxParallel(getMaxParallel());
+        logic.setSpeedBonus(getSpeedBonus());
+    }
+
+    @Override
+    public void addUIWidgets(ModularWindow.Builder builder, UIBuildContext buildContext) {
+        super.addUIWidgets(builder, buildContext);
+        if (debug) {
+            builder.widget(createTestButton(builder)).widget(createTestButton2()).widget(createTestButton3());
+        }
+    }
+
+    protected ButtonWidget createTestButton(IWidgetBuilder<?> builder) {
```

Comment:
Hope all 3 of these will get their names needed and you could just return Widget here instead of the type itself.